retis/src/core/filters/meta/filter.rs

```diff
@@ -391,75 +413,131 @@ impl FilterMeta {
+        let expr = filter.split(' ').collect::<Vec<_>>();
+
+        let [lhs, op, rhs]: [&str; 3] = match expr.len() {
+            3 => expr
+                .try_into()
+                .map_err(|_| anyhow!("cannot split filter ({filter})"))?,
+            1 => [expr[0], "!=", "0"],
+            _ => bail!("invalid filter ({filter})"),
+        };
+
+        let lhs: Vec<_> = lhs
+            .split('.')
+            .map(|x| {
+                let mut elem = x.split(':');
+                // member is mandatory.
+                let member = elem.next().ok_or_else(|| anyhow!("member is mandatory"))?;
+
+                // mask is optional and must be a number.
+                // Can be under the form [~]hex_num
+                let mask = if let Some(el) = elem.next() {
+                    Self::parse_mask(el)?
+                } else {
+                    0x0
+                };
+
+                // tgt_type is optional.
+                let tgt_type = elem.next();
+
+                Ok(LhsNode {
+                    member,
+                    mask,
+                    tgt_type,
+                })
+            })
+            .collect::<Result<Vec<LhsNode<'_>>>>()?;
+
+        if lhs.len() <= 1 {
+            bail!("expression does not point to a member");
+        }
+
+        if lhs[0].member != "sk_buff" {
+            bail!("starting struct isn't supported (!= sk_buff)");
+        }
+
+        Ok((lhs, MetaCmp::from_str(op)?, rhs))
+    }
+
     pub(crate) fn from_string(fstring: String) -> Result<Self> {
-        let btf = &inspector()?.kernel.btf;
+        let btf_info = &inspector()?.kernel.btf;
         let mut ops: Vec<_> = Vec::new();
         let mut offt: u32 = 0;
         let mut stored_offset: u32 = 0;
         let mut stored_bf_size: u32 = 0;
+        let mut mask = 0;
 
-        // Check the correctness, perform preliminar checks for the
-        // operator type against the target (e.g. >= against number).
-        //
-        // Once the lmo type is known, compare the rval against the
-        // lmo type (e.g. INT against LONG, sign)
-        let re = Regex::new(MF_RE)?;
-
-        let Some((_, [lval, op, rval])) = re.captures(&fstring).map(|caps| caps.extract()) else {
-            bail!("Invalid filter expression.")
-        };
+        let (mut fields, op, rval) = Self::parse_filter(&fstring)?;
 
-        let mut fields: Vec<_> = lval.split('.').collect();
+        // At least two elements are present
+        let init_sym = fields.remove(0).member;
 
-        // The captures ensure at least two elements are present
-        let init_sym = fields.remove(0);
-
-        if !init_sym.eq("sk_buff") {
-            bail!("unsupported data structure {init_sym}. sk_buff must be used.")
-        }
-
-        let mut types = btf
+        let mut types = btf_info
             .resolve_types_by_name(init_sym)
             .map_err(|e| anyhow!("unable to resolve sk_buff data type {e}"))?;
```

Comment:
nit: Not part of this MR, but technically this could be another struct. The check for `sk_buff` is done elsewhere and could be extended.